plugins/arDominionB5Plugin/js/sidebarPaginatedList.js

```diff
@@ -0,0 +1,146 @@
+
+      // Stop execution when pagination is not needed
+      if (this.totalPages < 2) {
+        return;
+      }
+
+      var pageLinkBehavior = {
+        updatePageLinkState: function (disabled) {
+          this.parent().toggleClass("disabled", disabled);
+          this.attr("aria-disabled", disabled);
+        },
+      };
+      this.$prev = this.$element
+        .find(".page-link-prev")
+        .extend(pageLinkBehavior);
+      this.$next = this.$element
+        .find(".page-link-next")
+        .extend(pageLinkBehavior);
+
+      this.$prev.updatePageLinkState(true);
+
+      this.$pageInput = this.$element.find("input[type=number]");
+      this.$results = this.$element.find("> ul");
+      this.$spinner = this.$element.find(".spinner");
+      this.$resultStart = this.$element.find(".result-start");
+      this.$resultEnd = this.$element.find(".result-end");
+
+      this.init();
+    }
+    init() {
+      this.$next.on("click", this.next.bind(this));
+      this.$prev.on("click", this.prev.bind(this));
+      this.$pageInput.on("change", this.change.bind(this));
+    }
+    next(e) {
+      e.preventDefault();
+      this.fetchResults(this.currentPage + 1);
+    }
+    prev(e) {
+      e.preventDefault();
+      this.fetchResults(this.currentPage - 1);
+    }
+    // Setter/getter of busy state
+    busy(busy) {
+      // Getter
+      if (typeof busy === "undefined") {
+        return this._busy;
+      }
+
+      // Setter
+      this._busy = busy;
+
+      var $spinner = this.$spinner;
+
+      if (busy) {
+        this.$prev.updatePageLinkState(true);
+        this.$next.updatePageLinkState(true);
+
+        this.busyTimer && clearTimeout(this.busyTimer);
+        this.busyTimer = setTimeout(function () {
+          $spinner.removeClass("invisible");
+        }, BUSY_THRESHOLD);
+      } else {
+        clearTimeout(this.busyTimer);
+        $spinner.addClass("invisible");
+        this.$prev.updatePageLinkState(this.currentPage == 1);
+        this.$next.updatePageLinkState(this.currentPage == this.totalPages);
+      }
+    }
+    // Fetch items for a given page and inject the results in the DOM
+    fetchResults(page) {
+      if (this.busy()) {
+        return;
+      }
+
+      if (page < 1 || page > this.totalPages) {
+        this.$pageInput.prop("value", this.currentPage);
+        return;
+      }
+
+      $.ajax({
+        url: this.url,
+        type: "GET",
+        context: this,
+        dataType: "json",
+        data: { page: page },
+        beforeSend: function () {
+          this.busy(true);
+        },
+        success: function (data) {
+          this.currentPage = page;
+          this.$pageInput.prop("value", page);
+
+          this.$results.empty();
+
+          var len = data["results"].length;
+          for (var i = 0; i < len; i++) {
+            this.$results.append(
+              $('<li class="list-group-item px-2 py-1">').append(
```

Comment:
I didn't need those padding in the list group used in the aggregations. Maybe it's caused by the ul/li elements, or may just be that the links are missing `.list-group-item-action`.

Reply:
I was trying to condense the content a little but I'll update so it looks more like the facets.

Reply:
Fixed, using anchors with list-group-item and list-group-item-action.